website/content/docs/agent/index.mdx

```diff
@@ -94,12 +96,17 @@ configuration entries:
 
 #### retry Stanza
 
-The `vault` stanza may contain a `retry` stanza that controls how failing
-Vault requests are handled, whether these requests are issued in order to render
+The `vault` stanza may contain a `retry` stanza that controls how failing Vault
+requests are handled, whether these requests are issued in order to render
 templates, or are proxied requests coming from the proxy cache subsystem.
 Auto-auth, however, has its own notion of retrying and is not affected by this
 section.
 
+For requests from the templating engine, Agent will restart its internal runner
```

Comment:
This gets into details (e.g. "its internal runner") that the user should not need to know about. I also don't think I confidently understand what's going to happen. Does "perform retries again once all retries are exhausted" mean that if I have`num_retries` at 5, I get 5 retries from the `retry` stanza and another 5 from the template rendering?

Reply:
It means that the templating engine will restart again with a count=0. It's mostly a backoff reset that doesn't cause agent to exit unless explicitly stated to do so. The scenario you gave is more of chained retries which _can_ happen if templating is going through the cache, but that's different from what's being described here.

The internal runner might be too much detail, though. What's a good way to describe this behavior without having to pull back the curtain too much on how it works under the hood?

Reply:
Removed the mentions of the internal runner.